NeuroID/src/main/java/com/neuroid/tracker/callbacks/NIDSensorHelper.kt

```diff
@@ -34,65 +34,76 @@ object NIDSensorHelper {
                     nidSensors.accelerometer.status = NIDSensorStatus.AVAILABLE
                 }
             }
-            NIDLog.i(TAG, "Sensor:${sensor.name} ${sensor.type}")
+            logger.i(TAG, "Sensor:${sensor.name} ${sensor.type}")
         }
 
-        restartSensors()
+        restartSensors(nSensors)
     }
 
     private fun initSensorManager(context: Context) {
         if (sensorManager == null)
             sensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
     }
 
+    @VisibleForTesting
+    // only called by unit tests, do not call outside of tests
+    internal fun resetSensorManager() {
+        sensorManager = null
+    }
+
+
     fun stopSensors() {
         sensorManager?.unregisterListener(listener)
     }
 
-    fun restartSensors() {
-        listener = NIDSensorGenListener {
+    fun restartSensors(nSensors: NIDSensors = nidSensors) {
+        listener = getNIDGenListener(nSensors, firstValuesGyro, firstValuesAccel)
+
+        gyroscopeSensor?.let {
+            sensorManager?.registerListener(listener, it, 10_000, 10_000)
+        }
+        accelerometerSensor?.let {
+            sensorManager?.registerListener(listener, it, 10_000, 10_000)
+        }
+    }
+
+    @VisibleForTesting
```

Comment:
Just curious, what does this tag do? As far as I could tell making it internal allowed testing

Reply:
Tells folks that this is made visible for testing purposes. It is not private so SDK developers can still hit this.